pipeline/pipeline.go

```diff
@@ -21,6 +22,14 @@ func (p *Pipeline) Submit(in *stage.Input, vr *validators.Request) {
 		return
 	}
 	vr.URL = url
+	if vr.Metadata != nil {
```

Comment:
I wonder if we should do this in the http handler so that we can return an error the client.